### ISSUE TITLE
[Bug]: proxy not working using chromium

### ISSUE THREAD

**Issue by gabrieImoreira** [microsoft/playwright-python#2517]
### Version

1.41.1

### Steps to reproduce

I have this code:

```
from playwright.async_api import async_playwright
import asyncio

async def main():
    async with async_playwright() as playwright:
        my_proxy = {"server": "http://p.webshare.io:80", "username": my_username, "password": my_passwd}
        browser = await playwright.chromium.launch(proxy={"server": "per-context"})
        context = await browser.new_context(proxy=my_proxy)
        page = await context.new_page()

        await page.goto("http://lumtest.com/myip.json")
        html_content = await page.content()
        print(html_content)

        await context.close()
        await browser.close()

asyncio.run(main())
```

### Expected behavior

My proxy works

### Actual behavior

Proxy doesnt't work

### Additional context

When I try to check my IP on http://lumtest.com/myip.json or other website with my code, my proxy doesn't work. I tried several variations, for example, pass proxy on launch, use other proxy, but I came to the conclusion that the problem is with the playwright. The only solution I found for my case is to change ```playwright.chromium.launch``` to ```playwright.webkit.launch```, but it doesn't work for me because a I need to configure a chromium browser.

I tried the same thing with a docker container (ubuntu) and I got the same result... I get no error message, just that my proxy doesn't work.

I did a lot of research on the internet, forums, official documentation, but the only configuration provided is this one. https://playwright.dev/python/docs/network

### Environment

```Text
- Operating System: [Windows 11, Linux]
- CPU: [x64]
- Browser: [Chromium]
- Python Version: [3.10, 3.11]
- Other info:
```

**Comment by mxschmitt** [comment]
I tried to reproduce and using [this squid proxy](https://hub.docker.com/r/ubuntu/squid) on macOS and for me the request was going through it. (checked the logs)

We fixed a similar issue in v1.46 which we'll release in a few days. You can either wait for it or use Google Chrome beta in the meantime, which should have the fix included.

**Comment by mxschmitt** [comment]
I'll close it for now since we weren't able to reproduce. Feel free to re-file with more details.
